colossalai/zero/gemini/placement_policy.py

```diff
@@ -95,6 +107,24 @@ def setup_grads_device(
         self.keep_gathered_chunk_mem = total_chunk_mem * (1 - self.shard_param_frac)
         self.keep_cuda_chunk_mem = total_chunk_mem * (1 - self.offload_param_frac)
 
+    def get_prefetch_chunks(
+        self, is_warmup: bool, compute_list: tuple, compute_idx: int, async_works: Dict[Chunk, dist.Work]
+    ) -> List[Chunk]:
+        if is_warmup:  # no prefetch during warmup since we need compute_list
+            return []
+        can_prefetch = self.max_prefetch - len(async_works)
+        prefetch = []
+        for i in range(compute_idx + 1, len(compute_list)):
+            for chunk in compute_list[i]:
+                if len(prefetch) >= can_prefetch:
+                    break
+                if chunk not in prefetch and chunk not in self.chunk_manager.accessed_chunks:
+                    prefetch.append(chunk)
+            else:
```

Comment:
Where its pair `if`?

Reply:
`else` match with `for` loop, to break out of the outer for loop